pinot-core/src/main/java/org/apache/pinot/core/query/executor/ServerQueryExecutorV1Impl.java

```diff
@@ -441,7 +444,15 @@ private InstanceResponseBlock executeInternal(TableDataManager tableDataManager,
       if (queryContext.isExplain()) {
         instanceResponse = getExplainResponseForNoMatchingSegment(numTotalSegments, queryContext);
       } else {
-        instanceResponse = new InstanceResponseBlock(ResultsBlockUtils.buildEmptyQueryResults(queryContext));
+        if (QueryContextUtils.isTimeSeriesQuery(queryContext)) {
+          // TODO: handle invalid segments
+          TimeSeriesBlock seriesBlock = new TimeSeriesBlock(
+              queryContext.getTimeSeriesContext().getTimeBuckets(), Collections.emptyMap());
+          TimeSeriesResultsBlock resultsBlock = new TimeSeriesResultsBlock(seriesBlock);
+          instanceResponse = new InstanceResponseBlock(resultsBlock);
```

Comment:
Shouldn't this code be in `ResultsBlockUtils.buildEmptyQueryResults`?

Reply:
BTW, it looks like we have no tests verifying this code is called.

Reply:
Yeah we could move it there. I can raise a follow-up PR if required. Lmk @gortiz 

re: tests, that's expected. We are trying to get a Quickstart working right now. Once the Part-4 PR is merged we will start raising smaller PRs with tests, bug fixes and other improvements.

Test coverage should be fixed in October.